codec-http/src/main/java/io/netty/handler/codec/http/multipart/DiskFileUpload.java

```diff
@@ -157,8 +160,7 @@ protected String getBaseDirectory() {
 
     @Override
     protected String getDiskFilename() {
-        File file = new File(filename);
-        return file.getName();
+        return String.valueOf(fileNameCounter.incrementAndGet());
```

Comment:
@guidomedina unfortunately not true, http://docs.oracle.com/javase/7/docs/api/java/io/File.html#createTempFile%28java.lang.String,%20java.lang.String,%20java.io.File%29 "This method provides only part of a temporary-file facility. To arrange for a file created by this method to be deleted automatically, use the deleteOnExit() method.". Even after deleteOnExit is invoked the JVM must exit "gracefully" in order for the shutdown hook to delete the file.